### PR TITLE
Consolidate `ExtendByteSlice` implementations

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
# Description

This PR merges all the implementations of `extendByteSlice` helper functions into a single one, since they are the same.

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

File: helper/common/common.go [hunk 1]
```diff
@@ -174,3 +174,21 @@ func PadLeftOrTrim(bb []byte, size int) []byte {
 
 	return tmp
 }
+
+// ExtendByteSlice extends given byte slice by needLength parameter.
+// If trim set to true, it puts hard limit on length and capacity to needLength value.
+func ExtendByteSlice(b []byte, needLength int, trim bool) []byte {
```
Comment (ferranbt) [inline helper/common/common.go]: I would prefer that we do two functions for the extension instead, one for the trim and one without or even keep the one of the bitmap as it is. The reason is that `extendByteSlice` (as it is) is a useful function for Go high-performance programming and I do not want to have to worry about misusage.

**Review by ferranbt — CHANGES_REQUESTED** [review]

File: helper/common/common.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply helper/common/common.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1058/commits/f4fe15458c48131f939419d67a60747946d15d4e

File: helper/common/common.go [hunk 2]
```diff
@@ -174,3 +174,14 @@ func PadLeftOrTrim(bb []byte, size int) []byte {
 
 	return tmp
 }
+
+// ExtendByteSlice extends given byte slice by needLength parameter and trims it
+func ExtendByteSlice(b []byte, needLength int) []byte {
```
Comment (ferranbt) [inline helper/common/common.go:179]: Could you add a couple of unit tests for the function?

**Review by ferranbt — APPROVED** [review]

File: helper/common/common.go (hunk 2 above)
Comment (Stefan-Ethernal) [reply helper/common/common.go:179]: :heavy_check_mark: https://github.com/0xPolygon/polygon-edge/pull/1058/commits/b776177b72378fdc6a2497a769b4f6c4fafd1058

**Pull request merged**